File: AWS_GUI_Dev/BAWGUI.Results/ViewModels/ViewModelBase.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BAWGUI.Results.ViewModels
{
    public class ViewModelBase : INotifyPropertyChanged
    {
        /// <summary>
        /// Raise property changed event
        /// </summary>
        /// <param name="sender">The event sender</param>
        /// <param name="e">The event</param>
        //public event PropertyChangedEventHandler INotifyPropertyChanged.PropertyChanged;
	    public delegate void PropertyChangedEventHandler(object sender, PropertyChangedEventArgs e);

        public event System.ComponentModel.PropertyChangedEventHandler PropertyChanged;

        protected virtual void OnPropertyChanged([System.Runtime.CompilerServices.CallerMemberName()] string propertyName = null)
        {
            if (PropertyChanged != null)
            {
                PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
            }
        }
        protected virtual bool CanExecute(object param)
        {
            return true;
        }
    }
}

File: AWS_GUI_Dev/BAWGUI.Settings/ViewModels/ViewModelBase.vb
Imports System.ComponentModel

Namespace ViewModels
    Public Class ViewModelBase
        Implements INotifyPropertyChanged
        ''' <summary>
        ''' Raise property changed event
        ''' </summary>
        ''' <param name="sender">The event sender</param>
        ''' <param name="e">The event</param>
        Public Event PropertyChanged(sender As Object, e As PropertyChangedEventArgs) Implements INotifyPropertyChanged.PropertyChanged
        Protected Overridable Sub OnPropertyChanged(<Runtime.CompilerServices.CallerMemberName> Optional propertyName As String = Nothing)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
        End Sub
        Protected Overridable Function CanExecute(ByVal param As Object) As Boolean
            Return True
        End Function
    End Class
End Namespace
